src/systems/userinput/resolve-action-sets.js

```diff
@@ -1,150 +1,142 @@
+          leftHand.hovered.components["super-spawner"]))
+  );
+  userinput.toggleSet(
+    sets.leftHandHoveringOnVideo,
+    !leftHand.held && leftHand.hovered && leftHand.hovered.components["media-video"]
+  );
+  userinput.toggleSet(
+    sets.leftHandHoldingPen,
+    leftHand.held && leftHand.held.components.tags && leftHand.held.components.tags.data.isPen
+  );
+  userinput.toggleSet(sets.leftHandHoldingCamera, leftHand.held && leftHand.held.components["camera-tool"]);
+  userinput.toggleSet(sets.leftHandHoldingInteractable, leftHand.held);
 
-  const cursorHoveringOnInteractable =
-    cursorCanHover && cursorHand.get("hover-start").matches(".interactable, .interactable *");
-  const cursorHoveringOnCamera = cursorCanHover && cursorHand.get("hover-start").matches(".icamera, .icamera *");
-  const cursorHoveringOnPen = cursorCanHover && cursorHand.get("hover-start").matches(".pen, .pen *");
-  const cursorHoveringOnVideo = cursorCanHover && cursorHand.get("hover-start").components["media-video"];
-  const cursorHoldingInteractable =
-    !rightHandTeleporting &&
-    cursorHand.has("grab-start") &&
-    cursorHand.get("grab-start").matches(".interactable, .interactable *");
-  const cursorHoldingPen =
-    !rightHandTeleporting && cursorHand.has("grab-start") && cursorHand.get("grab-start").matches(".pen, .pen *");
+  userinput.toggleSet(sets.rightHandHoveringOnNothing, !rightHand.held && !rightHand.hovered);
+  userinput.toggleSet(
+    sets.rightHandHoveringOnPen,
+    !rightHand.held &&
+      rightHand.hovered &&
+      rightHand.hovered.components.tags &&
+      rightHand.hovered.components.tags.data.isPen
+  );
+  userinput.toggleSet(
+    sets.rightHandHoveringOnCamera,
+    !rightHand.held && rightHand.hovered && rightHand.hovered.components["camera-tool"]
+  );
+  userinput.toggleSet(
+    sets.rightHandHoveringOnInteractable,
+    !rightHand.held &&
+      (rightHand.hovered &&
+        ((rightHand.hovered.components.tags && rightHand.hovered.components.tags.data.offersRemoteConstraint) ||
+          rightHand.hovered.components["super-spawner"]))
+  );
+  userinput.toggleSet(
+    sets.rightHandHoveringOnVideo,
+    !rightHand.held && rightHand.hovered && rightHand.hovered.components["media-video"]
+  );
+  userinput.toggleSet(
+    sets.rightHandHoldingPen,
+    rightHand.held && rightHand.held.components.tags && rightHand.held.components.tags.data.isPen
+  );
+  userinput.toggleSet(sets.rightHandHoldingCamera, rightHand.held && rightHand.held.components["camera-tool"]);
+  userinput.toggleSet(sets.rightHandHoldingInteractable, rightHand.held);
 
-  const cursorHoldingCamera =
-    !rightTeleporter.isTeleporting &&
-    cursorHand.has("grab-start") &&
-    cursorHand.get("grab-start").matches(".icamera, .icamera *");
+  userinput.toggleSet(
+    sets.cursorHoveringOnNothing,
+    !rightHand.held && !rightHand.hovered && !rightRemote.held && !rightRemote.hovered
```

Comment:
The contract for these action sets is pretty surprising. I only know it because I wrote them. Perhaps the interaction system should enforce these rules such that (for example) `rightRemote.hovered` is guaranteed to be `null` if anything is in your right hand or right remote, or your right hand is hovering on anything. That way this line becomes,
`userinput.toggleSet(sets.cursorHoveringOnNothing, !rightRemote.hovered)`